### PR TITLE
Controllers Branch: adding example mapping files

### PR DESCRIPTION
**Pull request by samcake**

### CODE REVIEW

File: interface/resources/controllers/mapping-config.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "mappings": [ 
+        "src": "./mapping-hydra.json",
+        "src": "./mapping-xbox.json",
+        {
+            "name": "example mapping for standard to js function",
+            "channels": [ {
+                    "from": "Standard.B",
+                    "to": {
+                        "type":"js",
+                        "function": "function(value){  print("Standard.B = " + value );}"
```
Comment (jherico) [inline interface/resources/controllers/mapping-config.json]: this json needs to be escaped.  
`print(\"Standard.b = \" ...`

File: interface/resources/controllers/mapping-config.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "name": "Full Mapping config including the standard hydra and gamepad and one more thing",
+    "mappings": [ 
+        "src": "./mapping-hydra.json",
```
Comment (jherico) [inline interface/resources/controllers/mapping-config.json]: non valid json....  you're using object notation in an array.

File: libraries/gpu/src/gpu/GLBackendTexture.cpp [hunk 3]
```diff
@@ -339,8 +339,6 @@ GLBackend::GLTexture* GLBackend::syncGPUObject(const Texture& texture) {
 
                 GLTexelFormat texelFormat = GLTexelFormat::evalGLTexelFormat(texture.getTexelFormat(), srcFormat);
 
-                auto semantic = texture.getTexelFormat().getSemantic();
```
Comment (jherico) [inline libraries/gpu/src/gpu/GLBackendTexture.cpp]: unrelated code?

File: libraries/input-plugins/src/input-plugins/SixenseManager.cpp [hunk 4]
```diff
@@ -68,10 +68,10 @@ const float DEFAULT_REACH_LENGTH = 1.5f;
 
 SixenseManager::SixenseManager() :
     InputDevice("Hydra"),
+    _reachLength(DEFAULT_REACH_LENGTH),
 #ifdef __APPLE__
-    _sixenseLibrary(NULL),
```
Comment (jherico) [inline libraries/input-plugins/src/input-plugins/SixenseManager.cpp]: unrelated code?

File: libraries/render-utils/src/DeferredGlobalLight.slh [hunk 5]
```diff
@@ -41,7 +41,7 @@ struct SphericalHarmonics {
 
 vec4 evalSphericalLight(SphericalHarmonics sh, vec3 direction ) {
 
-    vec3 dir = direction.xzy; // we don;t understand why yet but we need to use z as vertical axis?
```
Comment (jherico) [inline libraries/render-utils/src/DeferredGlobalLight.slh]: unrelated code?

File: libraries/render-utils/src/DeferredGlobalLight.slh [hunk 6]
```diff
@@ -125,7 +125,10 @@ vec3 evalSkyboxGlobalColor(mat4 invViewMat, float shadowAttenuation, vec3 positi
     vec4 fragEyeVector = invViewMat * vec4(-position, 0.0);
     vec3 fragEyeDir = normalize(fragEyeVector.xyz);
     
-    vec3 color = diffuse.rgb * evalSphericalLight(ambientSphere, fragNormal).xyz * getLightAmbientIntensity(light);
```
Comment (jherico) [inline libraries/render-utils/src/DeferredGlobalLight.slh]: unrelated code?

**Comment by hifi-gustavo** [comment]
Build console output for commit: c02d33c17cebb798fbf7bff43b9579c9b6419a30 available at: https://gist.github.com/7623d2c85473191c56ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.dmg

File: libraries/gpu/src/gpu/GLBackendTexture.cpp (hunk 3 above)
Comment (samcake) [inline libraries/gpu/src/gpu/GLBackendTexture.cpp]: i think it s a warning removed in upstream, i have pulled upstream once more

File: libraries/input-plugins/src/input-plugins/SixenseManager.cpp (hunk 4 above)
Comment (samcake) [inline libraries/input-plugins/src/input-plugins/SixenseManager.cpp]: i think it s a warning removed in upstream, i have pulled upstream once more

**Comment by hifi-gustavo** [comment]
Build console output for commit: 502cb90e027b2e3038f0842114bdb20634ab2fbe available at: https://gist.github.com/7623d2c85473191c56ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.dmg

File: interface/resources/controllers/mapping-config.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{
+    "name": "Full Mapping config including the standard hydra and gamepad and one more thing",
+    "mappings": [
+        { "src": "./mapping-hydra.json" },
```
Comment (jherico) [inline interface/resources/controllers/mapping-config.json]: I'll merge this, but I think that there's no real reason for this to be an object....  

```
"mapping": [
  "./mapping-hydra.json",
  {
    ...
  }
]

File: interface/resources/controllers/mapping-config.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+                    "from": "Standard.B",
+                    "to": {
+                        "type":"js",
+                        "function": "function(value){  print(\"Standard.B = \" + value );}"
+                    }
+                }, {
+                    "from": "Standard.B",
+                    "to": {
+                        "type":"js",
+                        "src": "http://www.theNextBigThing.com/hifiInputSignalHandler.js"
```
Comment (jherico) [inline interface/resources/controllers/mapping-config.json]: still has a network URL.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 297c3c9ed0dbacd7bcd3e9b82e6f98c69703451e available at: https://gist.github.com/7623d2c85473191c56ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3d607f406f99a349504b32aedbfacef4d5ec296d available at: https://gist.github.com/7623d2c85473191c56ea<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6048.dmg

**Comment by samcake** [comment]
Can you merge

**Comment by samcake** [comment]
Thanks
